### DISCUSSION TITLE
Welcome to the-steam-awards Discussions!

### DISCUSSION THREAD

**Discussion by quer** [quer/the-steam-awards#16] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by ghost** [comment]
Hello, I found this repository through reddit r/steambot, tested it and found it amazing how everything was through modules, even with little knowledge I managed to use it, I have only 5 accounts and it worked well in all of them, my respect for maintaining this repository, I intend to explore more modules in the future, thanks.

**Comment by stickerx** [comment]
Hello sir

can run this script on Heroku ???

**Reply by quer** [reply]
Hey, I have not tried it, but as it is able to run Nodejs, i am quite sure that you can.

**Reply by stickerx** [reply]
I am a newbie to coding sir I need help^to to do that ;'(

**Reply by quer** [reply]
as i do not know how it works on that, i have never tried it. I use: 
- digitalocean.com
- vultr.com

but from that i can tell, after a look on this page: 
https://www.heroku.com/nodejs
Just follow the steps. i cant support you in how to use that.
Only how to use my script

**Comment by stickerx** [comment]
hello sir

please give me an example of how to add multi-account on config.js

like that ????


var config = [];
config.push({
	steam_user: "exapmple",
	steam_pass: "example",
	sharedSecret: "example"

var config = [];
config.push({
	steam_user: "exapmple2",
	steam_pass: "example2",
	sharedSecret: "example2"
});
module.exports = config;

**Reply by gmmanonymus111** [reply]
You can do it either by writing every account detail one by one
```javascript
var config = [];
config.push({
        steam_user: "account1",
        steam_pass: "p4ss1",
        sharedSecret: "sh4r3d53cr371"
});
config.push({
        steam_user: "account2",
        steam_pass: "p4ss2",
        sharedSecret: "sh4r3d53cr372"
});
config.push({
        steam_user: "account3",
        steam_pass: "p4ss3",
        sharedSecret: "sh4r3d53cr373"
});
config.push({
        steam_user: "account4",
        steam_pass: "p4ss4",
        sharedSecret: "sh4r3d53cr374"
});
module.exports = config;
```
or you can push all data once like this:
```javascript
var config = [];
config.push({
        steam_user: "account1",
        steam_pass: "p4ss1",
        sharedSecret: "sh4r3d53cr371"
},{
        steam_user: "account2",
        steam_pass: "p4ss2",
        sharedSecret: "sh4r3d53cr372"
},{
        steam_user: "account3",
        steam_pass: "p4ss3",
        sharedSecret: "sh4r3d53cr373"
},{
        steam_user: "account4",
        steam_pass: "p4ss4",
        sharedSecret: "sh4r3d53cr374"
});
module.exports = config;
```
Just don't initialise `var config = [];` again, or you'll make your `config` array empty

**Comment by stickerx** [comment]
thank you so much, sir, and sorry I am a beginner at coding

1 question, please

-how I can use commands

like;

like Image
Clear Alias
Remove All Game From Wishlist
etc.....

**Reply by gmmanonymus111** [reply]
I'm not sure there are commands for that, only modules. You can add modules to the main js file, like
`modules.push(require('./modules/likeImage'));`
So, the top of `main.js` should look something like this:
```javascript
const config = require('./config');
const Steam = require('steam');
const SteamTotp = require('steam-totp');
const SteamWebLogOn = require('steam-weblogon');
const request = require('./request');
Steam.servers = require('./steamServer');
var modules = [];

//add moduels order
modules.push(require('./modules/likeImage'));

function loop(index) {
	runBot(index, loop);
    // end 
}
...
```
Now, head over to `/modules/likeImage.js` and edit `fileToLike` and `appId` to any file you want to like, e.g.: 
```javascript
var fileToLike = "1628371795";
var appid = "730";
```
You have to do something similart to this for every module you add to your `main.js` file. It is possible to add commands, like `!likeImage {appID},{imageID}`, but it requires some more coding to do. Best aproach would be to add a command array in your `main.js`, and push commands from included modules. Then you just need to handle chat messages, to extract the IDs. The easiest aproach is to hard-code the commands directly, but in this case, if the module is not loaded, your program most likely will crash.

**Comment by stickerx** [comment]
thank you so much 

work look like a magic xD

**Reply by gmmanonymus111** [reply]
Glad I could help ^^

**Comment by quer** [comment]
thank @gmmanonymus111, for the answer.!

**Reply by gmmanonymus111** [reply]
Glad I could help ^^

**Comment by Lounas225** [comment]
Hi there. Newbie to these modules but i got most of them working as intended. Really nice work thank you @quer 

I can't however figure out how to set up the modules to post game reviews one at the time, in a pase, where it will use up the request quote to steam. How do you achieve this?

**Reply by quer** [reply]
Hey, Thank for using the project ! :)
I haven't tested and use that module, since steam change the allowed reviews quote. As there was/is a problem whit spam account. 
so i know steam did try to make a fix at it. 

but are you saying that you cant make the module to work. Or that you to quick use the max request limit, to create game reviews ?

**Reply by Lounas225** [reply]
Thanks for the quick reply!

I have a long list of games to review, i added them to "GameRecommend_Add" module and it works fine but it hits the Steam request limit fast.

So i was wondering can one make it run so it pauses for X-amount of time between writing reviews?
(Basically an automated review bot that eats up the cooldown timer)

**Reply by quer** [reply]
if you just want a it to wait between each recommend, just a promise whit settimeout in. so then it will wait between each. Note if you run account in cluster. this will still work, but it will be runned up top of each other. 

add it after line 16 here :
https://github.com/quer/the-steam-awards/blob/Master/modules/GameRecommend_Add.js#LL16C40-L16C40
after ´await Make(recommendation);´

```js
await new Promise(resolve => setTimeout(resolve, X-amount));
```
where `X-amount` is the delay. Note that 1000 = 1 sec

( did not test the code but somefing like that )

**Reply by Lounas225** [reply]
I tested the code and it sure works. Thanks!

Also noticed however that it does not stop after failing to recommend (i.e game already reviewed), the delay only works if the recommendation was successfull.

Is there a way to edit the code so it stops between every recommend (success or fail but still force the delay between) ?

**Reply by quer** [reply]
you can just move the code line, you added above. the `await Make(recommendation);` so then it will always wait. then do the logic. So no mather if it works or not. i will wait.
